### ISSUE TITLE
style.visibility andHotSpots

### ISSUE THREAD

**Issue by Bruschetta** [mpetroff/pannellum#580]
I have a strange behaviour with hotspots and  style.visibility=visible/hidden.
To be mores specific let's make two examples.
Ii use this code:
```
viewer = pannellum.viewer('panorama', {
        "type": "equirectangular",
        "panorama": "test.jpg",
        "showControls": false,
        "autoLoad": true,
        "compass": true,
        //"hotSpotDebug": true,
        "hotSpots": [
	
        ]
    });
    for (var i = 0; i < loop; i++) {
        if (urlPresent[i] == 1) {
            viewer.addHotSpot({
                "id": eventId[i],
                "pitch": pitch[i],
                "yaw": yaws[i],
                "cssClass": "custom-hotspot",
                "createTooltipFunc": hotspot,
                "createTooltipArgs": "descrConcat[i]" ,
                "URL": url[i]
            });
    //Without URL
        } else {
            viewer.addHotSpot({
                "id": eventId[i],
                "pitch": pitch[i],
                "yaw": yaws[i],
                "cssClass": "custom-hotspot",
                "createTooltipFunc": hotspot,
                "createTooltipArgs":descrConcat[i]
            });
        }
    }
}

// Hot spot creation function
    function hotspot(hotSpotDiv, args) {
        hotSpotDiv.classList.add('custom-tooltip');
        var span = document.createElement('span');
        var res = args.split("-");
        span.id = (res[1].trim());
        span.innerHTML = (res[0].trim());
        hotSpotDiv.appendChild(span);
        span.style.width = span.scrollWidth - 20 + 'px';
        span.style.marginLeft = -(span.scrollWidth - hotSpotDiv.offsetWidth) / 2 + 'px';
        span.style.marginTop = -span.scrollHeight - 12 + 'px';
    }

```
While able to generate dinamically the hotspots i'm not able to modify dom with js calls like document.getElementById(span.id).style.visibility = visible/hidden;
Calling console.log(document.getElementById(span.id)) i see that the spanelement have been modified but nothing happens on the screen.

Otherwise,if i use this code:
```

viewer = pannellum.viewer('panorama', ﻿{
	"type": "equirectangular",
    "panorama": "test.jpg",
	"autoLoad": true,
	"compass": true,
    "hotSpots": [
        {
            "pitch": pitch[0],
            "yaw": yaws[0],
            "cssClass": "custom-hotspot",
            "createTooltipFunc": hotspot,
            "createTooltipArgs": eventDescr[0]
        },
        {
            "pitch": pitch[1],
            "yaw": yaws[1],
            "cssClass": "custom-hotspot",
            "createTooltipFunc": hotspot,
            "createTooltipArgs": eventDescr[1]
        },
        {
            "pitch": pitch[2],
            "yaw":yaws[2],
            "cssClass": "custom-hotspot",
            "createTooltipFunc": hotspot,
            "createTooltipArgs": eventDescr[2]
        }
    ]
});

}

// Hot spot creation function
function hotspot(hotSpotDiv, args) {
    hotSpotDiv.classList.add('custom-tooltip');
    var span = document.createElement('span');
    var res = args.split("-");
    span.id = (res[1].trim());
    span.innerHTML = (res[0].trim());
    hotSpotDiv.appendChild(span);
    span.style.width = span.scrollWidth - 20 + 'px';
    span.style.marginLeft = -(span.scrollWidth - hotSpotDiv.offsetWidth) / 2 + 'px';
    span.style.marginTop = -span.scrollHeight - 12 + 'px';
} 
```
Now with this code icannot generate hotspotsdinamically BUT i CAN modify dom with js calls like document.getElementById(span.id).style.visibility = visible/hidden;
Calling console.log(document.getElementById(span.id)) i see that the span element have been modified and modifies are visibile on the screen too.

Can you give me some hints and tell me why i have this behaviour? Is there any solutions that makes me able to generate hotspots dinamically AND hide / show hotspots on the  screen with JS?
Thanks for the time and answers

**Comment by mpetroff** [comment]
The same code is used internally to add hot spots whether they're created from the configuration or they're added dynamically, so I don't see any reason there would be different behavior. Are you sure you're not accidentally adding two elements with the same ID? Have you selected the `<span>` using your browser's element inspector instead of just the `console.log` call to see if the element on the screen was actually changed?

**Comment by Bruschetta** [comment]
Yes i have inspected all the span elements with element inspector (F12) and all of them have different IDs.
BUT when i add the hotspots dinamically, js call "document.getElementById(span.id).style.visibility = visible/hidden" is not working AND span element still do not have visibility = visible/hidden in the inline style. 
I really cannot understand why i get this behavior.

**Comment by Bruschetta** [comment]
I can confirm that even with arrays of data statically allocated ex
```
pitch[0]=30;
pitch[1]=1.596;
pitch[2]=2;
yaws[0]=-50;
yaws[1]=-60;
yaws[2]=-40;
eventId[0]=8;
eventId[1]=2;
eventId[2]=5;
descrConcat[0]="test1 - 8";
descrConcat[1]="test2 - 2";
descrConcat[2]="test3 - 5";
```
If hotspots are created with configurators i'm able to modify the DOM with JS, dinamically i'm unable to

**Comment by mpetroff** [comment]
I don't see any way this would happen unless you're somehow not selecting the correct element. If you post a link to an example that demonstrates this behavior, I can take a look. Otherwise, I'm not sure there's much I can do.

**Comment by Bruschetta** [comment]
Today i'm going to post two fiddle to show the behaviour i'm having.
Please  do not close the thread  :)

**Comment by Bruschetta** [comment]
I have created a fiddle in which i reproduce the error i get. 
Can you give me some hints for the problem? Why jquery is not changing the visibility of the SPAN?
https://jsfiddle.net/Lxu2q3pu/

**Comment by mpetroff** [comment]
In the example you provided, `document.getElementById('2').style.visibility = 'visible'` works fine for me in both Firefox and Chrome. I don't see what the issue is.

As for jQuery, I've never used it and have no intention of ever using it, so I can't answer any questions about it.

**Comment by Bruschetta** [comment]
First of all, thank you for the time you are spending in your answers.
When you say "document.getElementById('2').style.visibility = 'visible'" works fine for me in both Firefox and Chrome" are you telling me that in the fiddle you are able to see span with id 2? I'm using the same javascript command but on different browsers and indifferent machines i have the same behaviour, nothing is  shown unfortunately.

**Comment by Bruschetta** [comment]
Ok i was able to generate 2 fiddle. One that is working as intended, the one that have no dynamic hotspots and another that have dynamic hotspots and that not permit meto hide/show dom elements.
WORKING - NO DYNAMIC HOTSPOTS    https://jsfiddle.net/Lxu2q3pu/ 
NOT WORKING - DYNAMIC HOTSPOTS  https://jsfiddle.net/boLtzbvq/

**Comment by mpetroff** [comment]
With the new "not working" example, I was able to figure out what the problem was. As I suspected, you have multiple elements with the same `id`, and the visibility property was being set on the wrong one. This happens because you are dynamically adding the hot spots while the panorama is in the process of loading. Each hot spot is created immediately and then created again when the loading process finishes. If you trigger adding the hot spots with the an event listener on the `load` event, you won't have this issue.
